posts/2024-06-13-Rust-1.79.0.md

```diff
@@ -0,0 +1,167 @@
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.79.0 stable
+
+### Inline `const` expressions
+
+`const { ... }` blocks are now stable in expression position, permitting
+explicitly entering a const context without requiring extra declarations (e.g.,
+defining `const` items or associated constants on a trait).
+
+Unlike const items (`const ITEM: ... = ...`), inline consts are able to make
+use of in-scope generics, making them particularly useful for inline code snippets. For example, a pattern like:
+
+```rust
+const EMPTY: Option<Vec<u8>> = None;
+let foo = [EMPTY; 100];
+```
+
+can now be written like this:
+
+```rust
+let foo = [const { None }; 100];
+```
+
+Notably, this is also true of generic contexts, where previously a verbose trait declaration with an associated constant would be required:
+
+```rust
+fn create_none_array<T, const N: usize>() -> [Option<T>; N] {
+    [const { None::<T> }; N]
+}
+```
+
+This makes this code much more succinct and easier to read.
+
+See the [reference documentation](https://doc.rust-lang.org/nightly/reference/expressions/block-expr.html#const-blocks) for details.
+
+### Accept bounds in associated type position
+
+Rust 1.79 stabilizes the associated item bounds syntax, which allows us to put
+bounds in associated type position within other bounds, i.e.
+`T: Trait<Assoc: Bounds...>`. This avoids the need to provide an extra,
+explicit generic type just to constrain the associated type.
+
+This feature allows specifying bounds in a few places that previously either
+were not possible or imposed extra, unnecessary constraints on usage:
+
+* **`where` clauses** - in this position, this is equivalent to breaking up the bound into two (or more) `where` clauses. For example, `where T: Trait<Assoc: Bound>` is equivalent to `where T: Trait, <T as Trait>::Assoc: Bound`.
+* **Supertraits** - a bound specified via the new syntax is implied when the trait is used, unlike where clauses. Sample syntax: `trait CopyIterator: Iterator<Item: Copy> {}`.
+* **Associated type item bounds** - This allows constraining the *nested* rigid projections that are associated with a trait's associated types. e.g. `trait Trait { type Assoc: Trait2<Assoc2: Copy>; }`.
+* **opaque item bounds (RPIT, TAIT)** - This allows constraining associated types that are associated with the opaque without having to *name* the opaque. For example, `impl Iterator<Item: Copy>` defines an iterator whose item is `Copy` without having to actually name that item bound.
```

Comment:
This phrase makes me think “What's an opaque item?” Shouldn't this say “opaque type”, or is there a nuance here?

Reply:
Not sure - I think this is copied from the stabilization report. Cc @compiler-errors, maybe you can clarify?

Reply:
Sure, `s/opaque/opaque type` if that's clearer.